apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +125,38 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp extract_shipping_method_cost(shipping_method_id) do
+    shipping_method = Repo.get(ShippingMethod, shipping_method_id)
+    shipping_method.cost
+  end
+
+  defp calculate_package_total(
+         %{tax_total: tax_total, adjustment_total: adjustment_total, promo_total: promo_total},
+         shipping_cost
+       ) do
+    Enum.reduce([tax_total, adjustment_total, promo_total, shipping_cost], &Money.add!/2)
+  end
+
+  @doc """
+  Persists shipping_method_id to packages
+
+  Calculate pacakge total cost Sum(shipping_cost, adjustment_total,promo_total, shipping_cost)
+
+  Update package cost total in DB
+  """
+
+  @spec associate_package(Context.t()) :: Context.t()
```

Comment:
1. The packages are already associated (with the order). Come up with a better name.
2. The function spec does not match what the function actually returns, it returns a list of contexts. Why?
3. The `Enum.map` is dangerous as errors are not handled. Please take a look how `persist_shipment` safely traverses a list of repo operations, in a `Multi`.